development.yml

```diff
@@ -0,0 +1,10 @@
+version: "3.4"
+services:
+  beacon_node:
+    build:
+      target: development
+    image: lodestar:dev
+    volumes:
+      - ./:/usr/app
+      - ../dev/lodestar/node_modules:/usr/app/node_modules
```

Comment:
This `dev` folder is specific to your local environment directory structure? Could you make it more generic?

Reply:
umm, yes but it doesn't has any other significance other than separating out installs from the code base inside the container so as to not dirty the codebase in the IDE as well as save reinstall process if development image is rebuild. do suggest how to make it generic. with environment variable (from .env) may be?

Reply:
i will also be abstracting out lib files in the docker dev image (same reason as to not corrupt the IDE) plus docker cache use on the rebuilds (as the build artifacts will not corrupt main codebase)

Reply:
i will hookup docker persistent volume here as the only relevance is to prevent build artifacts in codebase

Reply:
generalized using docker volumes, also moved development.yml to docker folder.